packages/cli/src/commands/rules.ts

```diff
@@ -55,17 +57,26 @@ export default class RulesCommand extends ListingOutputAPICommand<Rule, RuleWith
 	sortKeyName = 'name'
 
 	protected getRulesByLocation = getRulesByLocation
-	protected listTableFieldDefinitions = ['name', 'roomId', 'locationId', 'locationName']
-	protected tableFieldDefinitions = tableFieldDefinitions
+	listTableFieldDefinitions = ['name', 'id', 'roomId', 'locationId', 'locationName']
+	tableFieldDefinitions = tableFieldDefinitions
 
 	async run(): Promise<void> {
 		const { args, argv, flags } = this.parse(RulesCommand)
 		await super.setup(args, argv, flags)
 
-		await this.processNormally(
-			args.idOrIndex,
-			() => { return this.getRulesByLocation(flags.locationId) },
-			(id) => { return this.client.rules.get(id, flags.locationId) },
+		await outputListing(this, args.idOrIndex,
+			() => this.getRulesByLocation(flags.locationId),
+			async id => {
+				if (flags.locationId) {
+					return this.client.rules.get(id, flags.locationId)
+				}
+				const allRules = await this.getRulesByLocation(flags.locationId)
+				const rule = allRules.find(rule => rule.id === id)
+				if (!rule) {
+					throw new CLIError(`could not find rule with id ${id}`)
+				}
+				return rule
+			},
```

Comment:
It would be a good refactor to extract this closure to a method/function like `getRulesByLocation` is.

Reply:
Good point...and it should be used in the `update` command which is almost certainly broken as well. I'm gonna make this pull request a draft and fix that too.

Reply:
(I guess I didn't remove roomId...I can do that though.)